pkg/subscriptions/subscriptions.go

```diff
@@ -385,3 +383,55 @@ func resolveSymbolicLink(path string) (string, error) {
 	}
 	return filepath.EvalSymlinks(path)
 }
+
+// mkdirAllIgnoreUmask creates a directory by ignoring the currently set umask.
+func mkdirAllIgnoreUmask(dir string, mode os.FileMode) error {
+	parent := dir
+	dirs := []string{}
+
+	// Find all parent directories which would have been created by MkdirAll
+	for {
+		if _, err := os.Stat(parent); os.IsNotExist(err) { //nolint: gocritic
```

Comment:
Which warning is this ignoring? At the very least document that here, so that future maintainers can notice an unexpected new warning.

Reply:
No error reported any more since I reduced the amount of else-ifs.